File: apps/backend/services/Identity/Identity.Presentation/Endpoints/Auth/Login/Login.Request.cs
namespace Identity.Presentation.Endpoints.Auth.Login;

/// <summary>
/// JSON body for POST /auth/login.
/// Kept separate from LoginCommand (Application layer) so that the HTTP contract
/// can evolve independently — e.g. renaming a field here doesn't affect the handler.
/// </summary>
public record LoginRequest(string Email, string Password);

File: apps/mobile/lib/core/features/authentication/models/requests/login_request.dart
/// Request body for POST /auth/login.
///
/// The Identity API expects:
///   { "email": "...", "password": "..." }
class LoginRequest {
  final String email;
  final String password;

  LoginRequest({required this.email, required this.password});

  Map<String, dynamic> toJson() => {
        'email': email,
        'password': password,
      };
}
